R/PackageInheritanceReporter.R

```diff
@@ -0,0 +1,212 @@
+#'   \href{https://adv-r.hadley.nz/r6.html}{\emph{Advanced R}'s second edition}.
+#'
+#' @section Public Methods:
+#' \describe{
+#'     \item{\code{set_package(pkg_name, pkg_path)}}{
+#'         \itemize{
+#'             \item{Set properties of this reporter. If pkg_name overrides a
+#'                 previously-set package name, any cached data will be removed.}
+#'             \item{\bold{Args:}}{
+#'                 \itemize{
+#'                 \item{\bold{\code{pkg_name}}: String with the name of the package.}
+#'                 \item{\bold{\code{pkg_path}}: Optional directory path to source
+#'                   code of the package. It is used for calculating test coverage.
+#'                   It can be an absolute or relative path.}
+#'                }
+#'             }
+#'         }
+#'     }
+#' }
+#' @importFrom R6 R6Class is.R6Class
+#' @importFrom DT datatable formatRound
+#' @importFrom data.table data.table rbindlist
+#' @importFrom methods is
+#' @export
+InheritanceReporter <- R6::R6Class(
+    "InheritanceReporter",
+    inherit = AbstractGraphReporter,
+
+    public = list(
+        get_summary_view = function(){
+
+            # Calculate network measures if not already done
+            # since we want the node measures in summary
+            invisible(self$network_measures)
+
+            # Create DT for display
+            tableObj <- DT::datatable(
+                data = self$nodes
+                , rownames = FALSE
+                , options = list(
+                    searching = FALSE
+                    , pageLength = 50
+                    , lengthChange = FALSE
+                )
+            )
+            # Round the double columns to three digits for formatting reasons
+            numCols <- names(which(unlist(lapply(tableObj$x$data, is.double))))
+            tableObj <- DT::formatRound(columns = numCols, table = tableObj
+                                        , digits=3)
+            return(tableObj)
+        }
+    ),
+
+    active = list(
+        nodes = function(){
+            if (is.null(private$cache$nodes)){
+                log_info("Extracting classes as nodes...")
+
+                pkg_env <- private$get_pkg_env()
+
+                nodeList <- list(data.table::data.table(
+                    node = character(0), classType = character(0)
+                ))
+
+                for (thisObjName in names(pkg_env)) {
+
+                    thisObj <- get(thisObjName, pkg_env)
+
+                    # S4 classes and References classes
+                    # Specified class name is the important name
+                    if (grepl("^\\.__C__", thisObjName) & isS4(thisObj)) {
+
+                        if (methods::is(thisObj, "refClassRepresentation")) {
+                            thisObjClassType <- "Reference"
+                        } else {
+                            thisObjClassType <- "S4"
+                        }
```

Comment:
minor thing but I like to avoid one-branch if statements when the logic inside is this simple. I think this is easier to read:

```
thisObjClassType <- "S4"
if (methods::is(thisObj, "refClassRepresentation")) {
    thisObjClassType <- "Reference"
}
```

Where basically you have a default assumption and then selectively override it.